src/full.rs

```diff
@@ -0,0 +1,157 @@
+use crate::{Body, SizeHint};
+use bytes::{Buf, Bytes};
+use http::HeaderMap;
+use std::borrow::Cow;
+use std::convert::{Infallible, TryFrom};
+use std::marker::Unpin;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// A body that consists of a single chunk.
+#[derive(Clone, Copy, Debug)]
+pub struct Full<D> {
+    data: Option<D>,
+}
+
+impl<D> Full<D>
+where
+    D: Buf + Unpin,
+{
+    /// Create a new `Full`.
+    pub fn new(data: D) -> Self {
+        let data = if data.has_remaining() {
+            Some(data)
+        } else {
+            None
+        };
+        Full { data }
+    }
+}
+
+impl<D> Body for Full<D>
+where
+    D: Buf + Unpin,
+{
+    type Data = D;
+    type Error = Infallible;
+
+    fn poll_data(
+        mut self: Pin<&mut Self>,
+        _cx: &mut Context<'_>,
+    ) -> Poll<Option<Result<D, Self::Error>>> {
+        Poll::Ready(self.data.take().map(Ok))
+    }
+
+    fn poll_trailers(
+        self: Pin<&mut Self>,
+        _cx: &mut Context<'_>,
+    ) -> Poll<Result<Option<HeaderMap>, Self::Error>> {
+        Poll::Ready(Ok(None))
+    }
+
+    fn is_end_stream(&self) -> bool {
+        self.data.is_none()
+    }
+
+    fn size_hint(&self) -> SizeHint {
+        self.data
+            .as_ref()
+            .map(|data| {
+                u64::try_from(data.remaining())
+                    .map(SizeHint::with_exact)
+                    .unwrap_or_else(|_| {
+                        let mut hint = SizeHint::new();
+                        hint.set_lower(u64::MAX);
```

Comment:
It looks like this is saying that if, say on some target where `usize` is bigger than `u64`, and the length overflows a `u64`, then we figure the lower hint is just _at least_ the max `u64`, yea?

Reply:
Exactly. But I admit it may be overcautious. Should we just `.unwrap()` here, instead?

Reply:
imo an `unwrap` is probably fine. If your body size is larger than u64 you probably have bigger problems 😜

Reply:
OK, I've replaced it with `unwrap`.